File: Gauss.java
public class Gauss {
    public static void main(String[] args) {
        double[][] A = {{1, 3, 7}, {3, 4, 1}, {4, 9, 1}};
        double[] b = {3, 1, 9};
        double[] x = gauss(A, b);
        for (double value : x) {
            System.out.print(value + " ");
        }
    }

    public static double[] gauss(double[][] A, double[] b) {
        int n = A.length;

        // Vorwärtselimination
        for (int k = 0; k < n - 1; k++) {
            for (int i = k + 1; i < n; i++) {
                double factor = A[i][k] / A[k][k];
                for (int j = k + 1; j < n; j++) {
                    A[i][j] -= factor * A[k][j];
                }
                b[i] -= factor * b[k];
            }
        }

        // Rückwärtssubstitution
        double[] x = new double[n];
        x[n - 1] = b[n - 1] / A[n - 1][n - 1];
        for (int i = n - 2; i >= 0; i--) {
            double s = b[i];
            for (int j = i + 1; j < n; j++) {
                s -= A[i][j] * x[j];
            }
            x[i] = s / A[i][i];
        }

        return x;
    }
}

File: gauss.py
def gauss(A, b):
    n = len(A)

    # Vorwärtselimination
    for k in range(n-1):
        for i in range(k+1, n):
            factor = A[i][k] / A[k][k]
            for j in range(k+1, n):
                A[i][j] -= factor * A[k][j]
            b[i] -= factor * b[k]

    # Rückwärtssubstitution
    x = [0.0] * n
    x[n-1] = b[n-1] / A[n-1][n-1]
    for i in range(n-2, -1, -1):
        s = b[i]
        for j in range(i+1, n):
            s -= A[i][j] * x[j]
        x[i] = s / A[i][i]

    return x


A = [[1, 3, 7], [3, 4, 1], [4, 9, 1]]
b = [3, 1, 9]
x = gauss(A, b)
print(x)
